aspnetcore/web-api/action-return-types.md

```diff
@@ -10,61 +11,41 @@ uid: web-api/action-return-types
-
 * [Specific type](#specific-type)
 * [IActionResult](#iactionresult-type)
 * [ActionResult\<T>](#actionresultt-type)
 
-::: moniker-end
-
-::: moniker range="<= aspnetcore-2.0"
-
-* [Specific type](#specific-type)
-* [IActionResult](#iactionresult-type)
-
-::: moniker-end
-
 This document explains when it's most appropriate to use each return type.
 
 ## Specific type
 
 The simplest action returns a primitive or complex data type (for example, `string` or a custom object type). Consider the following action, which returns a collection of custom `Product` objects:
 
-[!code-csharp[](../web-api/action-return-types/samples/2x/WebApiSample.Api.21/Controllers/ProductsController.cs?name=snippet_Get)]
+[!code-csharp[](../web-api/action-return-types/samples/3.x/WebApiSample.Api.31/Controllers/ProductsController.cs?name=snippet_Get)]
 
 Without known conditions to safeguard against during action execution, returning a specific type could suffice. The preceding action accepts no parameters, so parameter constraints validation isn't needed.
 
 When multiple return types are possible, it's common to mix an <xref:Microsoft.AspNetCore.Mvc.ActionResult> return type with the primitive or complex return type. Either [IActionResult](#iactionresult-type) or [ActionResult\<T>](#actionresultt-type) are necessary to accommodate this type of action. Several samples of multiple return types are provided in this document.
 
 ### Return IEnumerable\<T> or IAsyncEnumerable\<T>
 
-In ASP.NET Core 2.2 and earlier, returning <xref:System.Collections.Generic.IEnumerable%601> from an action results in synchronous collection iteration by the serializer. The result is the blocking of calls and a potential for thread pool starvation. To illustrate, imagine that Entity Framework (EF) Core is being used for the web API's data access needs. The following action's return type is synchronously enumerated during serialization:
+Returning <xref:System.Collections.Generic.IEnumerable%601> from an action results in synchronous collection iteration by the serializer. The result is the blocking of calls and a potential for thread pool starvation. To illustrate, imagine that Entity Framework (EF) Core is being used for the web API's data access needs. The following action's return type is synchronously enumerated during serialization:
 
 ```csharp
 public IEnumerable<Product> GetOnSaleProducts() =>
     _context.Products.Where(p => p.IsOnSale);
 ```
 
-To avoid synchronous enumeration and blocking waits on the database in ASP.NET Core 2.2 and earlier, invoke `ToListAsync`:
-
-```csharp
-public async Task<IEnumerable<Product>> GetOnSaleProducts() =>
-    await _context.Products.Where(p => p.IsOnSale).ToListAsync();
-```
-
-In ASP.NET Core 3.0 and later, returning `IAsyncEnumerable<T>` from an action:
+Returning `IAsyncEnumerable<T>` from an action:
 
 * No longer results in synchronous iteration.
 * Becomes as efficient as returning <xref:System.Collections.Generic.IEnumerable%601>.
```

Comment:
Should we change this slightly? The "no longer" is comparing it to 2.x, I think.

Reply:
Yes ... I didn't really read every line of these .. **_obvously_** 🙈 As usual, I have *a gazillion* Blazor things to get back to. 😄

I'll remove that line.